website/docs/r/automation_connection.html.markdown

```diff
@@ -0,0 +1,95 @@
+---
+subcategory: "Automation"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_automation_connection"
+description: |-
+  Manages a Automation Connection.
+---
+
+# azurerm_automation_connection
+
+Manages a Automation Connection.
+
+## Example Usage
+
+```hcl
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "example" {
+  name     = "resourceGroup-example"
+  location = "West Europe"
+}
+
+data "azurerm_client_config" "example" {}
+
+resource "azurerm_automation_account" "example" {
+  name                = "account-example"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
+  sku {
+    name = "Basic"
+  }
+}
+
+resource "azurerm_automation_connection" "example" {
+  name                    = "connection-example"
+  resource_group_name     = azurerm_resource_group.test.name
+  automation_account_name = azurerm_automation_account.test.name
+  type                    = "AzureServicePrincipal"
+
+  values = {
+    "ApplicationId" : "00000000-0000-0000-0000-000000000000",
+    "TenantId" : data.azurerm_client_config.example.tenant_id,
+    "SubscriptionId" : data.azurerm_client_config.example.subscription_id,
+    "CertificateThumbprint" : "sample-certificate-thumbprint",
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Connection. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the resource group in which the Connection is created. Changing this forces a new resource to be created.
+
+* `automation_account_name` - (Required) The name of the automation account in which the Connection is created. Changing this forces a new resource to be created.
+
+* `type` - (Required) The type of the Connection - can be either builtin type `Azure`, `AzureClassicCertificate` `AzureServicePrincipal` or user custom types. Changing this forces a new resource to be created.
+
+* `values` - (Optional) A mapping of key value pairs passed to the connection. Different `type` needs different parameters in the `values`. Builtin types have required field values as below:
+
+  - `Azure`: parameters `AutomationCertificateName` and `SubscriptionID`.
+
+  - `AzureClassicCertificate`: parameters `SubscriptionName`, `SubscriptionId` and `CertificateAssetName`.
+
+  - `AzureServicePrincipal`: parameters `ApplicationId`, `CertificateThumbprint`, `SubscriptionId` and `TenantId`.
```

Comment:
I think it might be better to have 3 different resources with these properties exposed